Nightly Dotty workflow of 2024-11-13 failed

See https://github.com/scala/scala3/actions/runs/11809844416

`test_windows_full` failure:
```scala
Error:  Test dotty.tools.dotc.profile.TraceNameManglingTest.escapeBackslashes failed: java.lang.AssertionError: assertion failed: Missing 1 names [TraceEvent(file,\\.scala)] in events, got:
Error:  - file: [, .scala, Unit.class, package.class, Typeable$package.tasty, Predef.class, LowPriorityImplicits.class, Predef.tasty, RuntimeChecked.tasty, Annotation.class, Option.class, IterableOnce.class, Product.class, Equals.class, InlineParam.tasty, Boolean.class, LowPriorityImplicits2.class]
Error:  - typecheck: [class /\\, constructor <init>, variable /\\, parameter x$1, setter /\\_=, object /\\,  <none>, method /\\], took 1.929 sec
Error:      at scala.runtime.Scala3RunTime$.assertFailed(Scala3RunTime.scala:8)
Error:      at dotty.tools.dotc.profile.TraceNameManglingTest.checkTraceEvents$$anonfun$1(TraceNameManglingTest.scala:129)
Error:      at scala.runtime.function.JProcedure1.apply(JProcedure1.java:15)
Error:      at scala.runtime.function.JProcedure1.apply(JProcedure1.java:10)
Error:      at dotty.tools.dotc.profile.TraceNameManglingTest.compileWithTracer(TraceNameManglingTest.scala:113)
Error:      at dotty.tools.dotc.profile.TraceNameManglingTest.checkTraceEvents(TraceNameManglingTest.scala:118)
Error:      at dotty.tools.dotc.profile.TraceNameManglingTest.escapeBackslashes(TraceNameManglingTest.scala:49)
Error:      ...
[info] Test run dotty.tools.dotc.profile.TraceNameManglingTest finished: 1 failed, 0 ignored, 1 total, 1.934s
Picked up _JAVA_OPTIONS: -Xmx8192m -XX:ReservedCodeCacheSize=512m -XX:MaxMetaspaceSize=1024m
Error:  Failed: Total 675, Failed 1, Errors 0, Passed 674, Ignored 8
Error:  Failed tests:
Error:  	dotty.tools.dotc.profile.TraceNameManglingTest
Error:  (scala3-compiler-bootstrapped / Test / testOnly) sbt.TestsFailedException: Tests unsuccessful
Error:  Total time: 1938 s (32:18), completed Nov 13, 2024 4:42:48 AM
[info] Publishing build scan...
[info] https://develocity.scala-lang.org/s/kjkoutdaw5c32
Picked up _JAVA_OPTIONS: -Xmx8192m -XX:ReservedCodeCacheSize=512m -XX:MaxMetaspaceSize=1024m
Error: Process completed with exit code 1.
```

--- a/compiler/test/dotty/tools/dotc/profile/TraceNameManglingTest.scala
+++ b/compiler/test/dotty/tools/dotc/profile/TraceNameManglingTest.scala
@@ -27,7 +27,7 @@ class TraceNameManglingTest extends DottyTest {
   }
 
   @Test def escapeBackslashes(): Unit = {
-    val isWindows = sys.props("os.name").toLowerCase(Locale.ROOT).contains("windows")
+    val isWindows = sys.props("os.name").toLowerCase(Locale.ROOT).nn.contains("windows")
     val filename = if isWindows then "/.scala" else "\\.scala"
     checkTraceEvents(
       """

--- a/compiler/test/dotty/tools/dotc/profile/TraceNameManglingTest.scala
+++ b/compiler/test/dotty/tools/dotc/profile/TraceNameManglingTest.scala
@@ -46,7 +46,7 @@ class TraceNameManglingTest extends DottyTest {
         raw"setter /\\_="
       ).map(TraceEvent("typecheck", _))
         ++ Set(
-          TraceEvent("file", if isWindows then "/.scala" else "\\\\.scala")
+          TraceEvent("file", if isWindows then ".scala" else "\\\\.scala")
         )
     )
   }

--- a/compiler/test/dotty/tools/dotc/profile/TraceNameManglingTest.scala
+++ b/compiler/test/dotty/tools/dotc/profile/TraceNameManglingTest.scala
@@ -28,7 +28,8 @@ class TraceNameManglingTest extends DottyTest {
 
   @Test def escapeBackslashes(): Unit = {
     val isWindows = sys.props("os.name").toLowerCase(Locale.ROOT).nn.contains("windows")
-    val filename = if isWindows then "/.scala" else "\\.scala"
+    // It is not possible to create a file with backslash in name on Windows
+    val filename = if isWindows then "test.scala" else "\\.scala"
     checkTraceEvents(
       """
       |class /\ :
@@ -46,7 +47,8 @@ class TraceNameManglingTest extends DottyTest {
         raw"setter /\\_="
       ).map(TraceEvent("typecheck", _))
         ++ Set(
-          TraceEvent("file", if isWindows then "/.scala" else "\\\\.scala")
+          // See comment aboce for Windows limitations
+          TraceEvent("file", if isWindows then filename else "\\\\.scala")
         )
     )
   }

--- a/compiler/test/dotty/tools/dotc/profile/TraceNameManglingTest.scala
+++ b/compiler/test/dotty/tools/dotc/profile/TraceNameManglingTest.scala
@@ -27,7 +27,7 @@ class TraceNameManglingTest extends DottyTest {
   }
 
   @Test def escapeBackslashes(): Unit = {
-    val isWindows = sys.props("os.name").toLowerCase(Locale.ROOT) == "windows"
+    val isWindows = sys.props("os.name").toLowerCase(Locale.ROOT).contains("windows")
     val filename = if isWindows then "/.scala" else "\\.scala"
     checkTraceEvents(
       """
@@ -46,7 +46,7 @@ class TraceNameManglingTest extends DottyTest {
         raw"setter /\\_="
       ).map(TraceEvent("typecheck", _))
         ++ Set(
-          TraceEvent("file", if isWindows then ".scala" else "\\\\.scala")
+          TraceEvent("file", if isWindows then "/.scala" else "\\\\.scala")
         )
     )
   }

